pkg/operator/ceph/object/controller.go

```diff
@@ -137,6 +151,29 @@ func (c *ObjectStoreController) onUpdate(oldObj, newObj interface{}) {
 		return
 	}
 
+	// Check if the objectstore is being deleted. This code path is called when a finalizer is specified in the crd.
+	// When an objectstore is requested for deletion, K8s will only set the deletion timestamp if there are any finalizers in the list.
+	// K8s will only delete the crd and child resources when the finalizers have been removed from the crd.
+	if newStore.DeletionTimestamp != nil {
```

Comment:
```suggestion
	if !newStore.GetDeletionTimestamp().IsZero() {
```

It's just an implementation detail, feel free to use this or not.

Reply:
Sure; makes sense.